src/com/dostojic/climbers/controller/Controller.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
+package com.dostojic.climbers.controller;
+
+import com.dostojic.climbers.exception.LoginException;
+import com.dostojic.climbers.domain.Climber;
+import com.dostojic.climbers.logic.ClimberRepository;
+import com.dostojic.climbers.logic.CompetitionRepository;
+import com.dostojic.climbers.domain.User;
+import com.dostojic.climbers.logic.UserRepository;
+import java.util.List;
+import java.util.Optional;
+
+/**
+ *
+ * @author planina
+ */
+public class Controller {
+
+    private final ClimberRepository climberRepository;
+    private final CompetitionRepository competitionRepository;
+    private final UserRepository userRepository;
+
+//    private static Controller instance;
+    
+/*    private Controller() {
+        climberRepository = new ClimberRepositoryInMemoryImpl();
+        competitionRepository = new CompetitionRepositoryInMemoryImpl();
+        userRepository = new UserRepositoryInMemoryImpl();
+    }
+*/
+    public Controller(ClimberRepository climberRepository, CompetitionRepository competitionRepository, UserRepository userRepository) {
+        this.climberRepository = climberRepository;
+        this.competitionRepository = competitionRepository;
+        this.userRepository = userRepository;
+    }
+    
+    
+    
+//    public static Controller getInstance() {
+//        if (instance == null) {
+//            instance = new Controller();
+//        }
+//        return instance;
+//    }
```

Comment:
The aim is to create hexagonal architecture so we 
replaced singleton controller with constructor so we can inject dependencies. This way **Core** will not depend on the repository implementation (at the moment Controller only and in future System Operation classes).
That will be joined in [UserRepositoryDbbrImpl.java](https://github.com/dejanostojic/ps-climbers-server/pull/1/files#diff-c2eb14b89aad75c3d6191f177318c2c1a4a773014b89b1c43085ab17182d03d2)
or in **In memory adapters**, eg: [ClimberRepositoryInMemoryImpl](https://github.com/dejanostojic/ps-climbers-server/pull/1/files#diff-c55476c3c66fb5ef7a1c46c04859738369697073a7edbbcaffe5ca1444cd1b4e).
The wiring will be done in [MainFactory](https://github.com/dejanostojic/ps-climbers-server/pull/1/files#diff-6dfe23cf0c5da90720deeb63820d5cd6ada9aa7f6aeef502e4d98c919503dc2f)